src/libxtp/aomatrices/aomatrix.cc

```diff
@@ -105,18 +105,37 @@ void AOOverlap::Fill(const AOBasis& aobasis) {
 Eigen::MatrixXd AOOverlap::singleShellOverlap(const AOShell& shell) const {
   libint2::Shell::do_enforce_unit_normalization(false);
   libint2::Operator obtype = libint2::Operator::overlap;
-  libint2::Engine engine(obtype, shell.getSize(),
-                         static_cast<int>(shell.getL()), 0);
+  try {
+    libint2::Engine engine(obtype, shell.getSize(),
+                           static_cast<int>(shell.getL()), 0);
 
-  const libint2::Engine::target_ptr_vec& buf = engine.results();
-
-  libint2::Shell s = shell.LibintShell();
-  engine.compute(s, s);
-
-  Eigen::Map<const AOMatrix::MatrixLibInt> buf_mat(buf[0], shell.getNumFunc(),
-                                                   shell.getNumFunc());
+    const libint2::Engine::target_ptr_vec& buf = engine.results();
 
-  return buf_mat;
+    libint2::Shell s = shell.LibintShell();
+    engine.compute(s, s);
+
+    Eigen::Map<const AOMatrix::MatrixLibInt> buf_mat(buf[0], shell.getNumFunc(),
+                                                     shell.getNumFunc());
+    return buf_mat;
+  } catch (const libint2::Engine::lmax_exceeded& error) {
+    std::cout << "\nA libint error occured:\n" << error.what() << std::endl;
+    std::cout
+        << "\nYou requested a computation for a shell with angular momentum "
+        << error.lmax_requested()
+        << ",\nbut your libint package only supports angular momenta upto "
+        << error.lmax_limit() - 1 << "." << std::endl;
+    std::cout
+        << "\nTo fix this error you will need to reinstall libint with "
+           "support\n"
+           "for higher angular momenta. If you installed your own libint it\n"
+           "should be reconfigured and installed with the option "
+           "--with-max-am=<maxAngularMomentum>.\n"
+           "If your installed libint with your OS package manager, you will\n"
+           "need to setup you own libint installation with the \n"
+           "--with-max-am=<maxAngularMomentum> option set."
+        << std::endl;
+    exit(EXIT_FAILURE);
```

Comment:
hey great error message, but could you throw a std::runtime_error instead of `exit`. The reason is that someone might want to catch that error at some point.